core/src/main/java/software/amazon/awssdk/core/ApiName.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core;
+
+import static software.amazon.awssdk.utils.Validate.notNull;
+
+/**
+ * Encapsulates the API name and version of a library built using the AWS SDK.
+ *
+ * See {@link SdkRequestOverrideConfig.Builder#addApiName(ApiName)}.
+ */
+public final class ApiName {
+    private final String name;
+    private final String version;
+
+    private ApiName(BuilderImpl b) {
+        this.name = notNull(b.name, "name must not be null");
+        this.version = notNull(b.version, "version must not be null");
```

Comment:
Could also use `paramNotNull(b.name, "name")`! This isn't a blocker or anything, just wanted to spread the knowledge if you hadn't seen that.